packages/amplify_core_plugin_interface/lib/amplify_datastore_category.dart

```diff
@@ -34,42 +33,48 @@ class DataStoreCategory {
       // in the `onAttachedToEngine` but rather in the `configure()
       await plugin.configureModelProvider(modelProvider: plugin.modelProvider);
     } else {
-      throw (errorMsg);
+      throw ("DataStore plugin was not added. You've already added one.");
     }
   }
 
   Future<List<T>> query<T extends Model>(ModelType<T> modelType,
       {QueryPredicate where,
       QueryPagination pagination,
       List<QuerySortBy> sortBy}) {
-    return plugins.length == 1
-        ? plugins[0].query(modelType,
-            where: where, pagination: pagination, sortBy: sortBy)
-        : throw (errorMsg);
+    return _select((it) => it.query(modelType,
+        where: where, pagination: pagination, sortBy: sortBy));
   }
 
   Future<void> delete<T extends Model>(T model) {
-    return plugins.length == 1 ? plugins[0].delete(model) : throw (errorMsg);
+    return _select((it) => it.delete(model));
   }
 
   Future<void> save<T extends Model>(T model) {
-    return plugins.length == 1 ? plugins[0].save(model) : throw (errorMsg);
+    return _select((it) => it.save(model));
   }
 
   Stream<SubscriptionEvent<T>> observe<T extends Model>(
       ModelType<T> modelType) {
+    // TODO: Howto use _stream()? It will return a Future<Stream<...>>,

```

Comment:
```suggestion
    // TODO: Howto use _select()? It will return a Future<Stream<...>>,
```

Reply:
You can await on it? `return await _select((it) => it.observe());`

Reply:
I thought so! VS Code told me I'd need to put `async` on the method signature, to become:
```dart
async Stream<SubscriptionEvent<T>> observe<T extends Model>(ModelType<T> modelType) {
      ...
}
```
Would that be acceptable?